proposed/http-client/http-client.md

```diff
@@ -60,17 +80,6 @@ interface ClientInterface
     /**
      * Sends a PSR-7 request and returns a PSR-7 response.
      *
-     * Every technically correct HTTP response MUST be returned as-is, even if it represents an HTTP
-     * error response or a redirect instruction. The only special case is 1xx responses, which MUST
-     * be assembled in the HTTP client.
-     *
-     * The client MAY do modifications to the Request before sending it. Because PSR-7 objects are
-     * immutable, one cannot assume that the object passed to ClientInterface::sendRequest() will be the same
-     * object that is actually sent. For example, the Request object that is returned by an exception MAY
-     * be a different object than the one passed to sendRequest, so comparison by reference (===) is not possible.
```

Comment:
i am not on the slack where you discussed, but i would see value in repeating (in condensed form) the most important properties here. the interface is much more likely to be seen by users than the full spec. what would you think of keeping an abbreviated version of this information on the interface?

Reply:
In concept I can see the value in doing so, but in this case I cannot think of any notes worth replicating.  Things like "the object may not be the same" are rather edge-case-y.

Reply:
i was mainly thinking about the first part that error responses are not exceptions. but if you think its unnecessary, i am ok with this PR.